component-studio/component-runtime-di/src/main/java/org/talend/sdk/component/runtime/di/schema/TaCoKitGuessSchema.java

```diff
@@ -183,6 +193,95 @@ public void guessInputComponentSchema() throws Exception {
         throw new Exception("There is no available schema found.");
     }
 
+    public void guessComponentSchema(final Schema incomingSchema, final String outgoingBranch)
+            throws Exception {
+        try {
+            final Collection<ServiceMeta> services = componentManager
```

Comment:
agree this now, but please consider this case:
for tck button click, there are two ways at least:
1. this way now we use here, use reflect to call, as studio call this in a new java process, so run in a new jvm.
2. call real rest api to the embeded componet server which is started by studio jvm(not sure). Example : Check connection button for studio metadata

image this case : user set proxy in studio preference or project setting, and for example , tjdbcinput with guess schema button, user expect to go through the proxy, but if use way 1, may not work, see https://jira.talendforge.org/browse/TUP-35447

just show the info, no need to fix here.